Results of primitives (#893)
realm/realm-dart: CHANGELOG.md

**This project is in the Beta stage. The API should be quite stable, but occasional breaking changes may be made.**

### Enhancements
* Support results of primitives, ie. `RealmResult<int>`.

>   Result of primitives do not support notifications and that is captured in the type system. ([#162](https://github.com/realm/realm-dart/issues/162))

Reviewer: ```suggestion
  Result of primitives do not support notifications. ([#162](https://github.com/realm/realm-dart/issues/162))
```
I suggest we go with this shorter description.

>   Result of primitives do not support notifications and that is captured in the type system. ([#162](https://github.com/realm/realm-dart/issues/162))

Author: You tell me 😄 .. We don't support notifications on list of primitives either. It goes back to this internal discussion: https://mongodb.slack.com/archives/C010R3CMP3N/p1644327500575779, but perhaps I misunderstood?

>   Result of primitives do not support notifications and that is captured in the type system. ([#162](https://github.com/realm/realm-dart/issues/162))

Author: I will try to remove this restriction, and on list as well.

>   Result of primitives do not support notifications and that is captured in the type system. ([#162](https://github.com/realm/realm-dart/issues/162))

Author: My mistake. We do support notifications. Has exposed for both lists and results, and added tests.

### Enhancements
* Added `MutableSubscriptionSet.removeByType` for removing subscriptions by their realm object type. (Issue [#317](https://github.com/realm/realm-dart/issues/317))

> * Support results of primitives, ie. `RealmResult<int>`. (Issue [#162](https://github.com/realm/realm-dart/issues/162))

Reviewer 2: ```suggestion
* Support `RealmResults` of primitive types (bool, int, string ...), ie. `RealmResults<int>`. (Issue [#162](https://github.com/realm/realm-dart/issues/162))
```

Author: I think this would be inconsistent. We have used the word primitives before in the CHANGELOG without defining it, and we often refer to the class `RealmList` as realm list(s) in free text.

Reviewer 2: It sounds better to me. But it is just a suggestion. It is not important. As you decide.

> * Support notifications on all managed realm lists, including list of primitives, ie. `RealmList<int>.changes` is supported. ([#893](https://github.com/realm/realm-dart/pull/893))

Reviewer 2: ```suggestion
* Support notifications on all managed `RealmList`s, including list of primitive types, for example: `RealmList<int>.changes` is supported. ([#893](https://github.com/realm/realm-dart/pull/893))
```
